testutil/util.go

```diff
@@ -0,0 +1,190 @@
+package testutil
+
+import (
+	"path/filepath"
+	"time"
+
+	tmos "github.com/tendermint/tendermint/libs/os"
+	"github.com/tendermint/tendermint/node"
+	"github.com/tendermint/tendermint/p2p"
+	pvm "github.com/tendermint/tendermint/privval"
+	"github.com/tendermint/tendermint/proxy"
+	"github.com/tendermint/tendermint/rpc/client/local"
+	"github.com/tendermint/tendermint/types"
+	tmtime "github.com/tendermint/tendermint/types/time"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server/api"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+)
+
+func startInProcess(cfg Config, val *Validator) error {
+	logger := val.Ctx.Logger
+	tmCfg := val.Ctx.Config
+	tmCfg.Instrumentation.Prometheus = false
+
+	db := dbm.NewMemDB()
+	genDocProvider := node.DefaultGenesisDocProviderFunc(tmCfg)
+
+	app := simapp.NewSimApp(
+		logger, db, nil, true, make(map[int64]bool),
+		tmCfg.RootDir, 0,
+		baseapp.SetPruning(storetypes.NewPruningOptionsFromString(val.AppConfig.Pruning)),
+		baseapp.SetMinGasPrices(cfg.MinGasPrices),
+	)
```

Comment:
Why can't we callow this to be configured in `Config`? In my mind it should be possible for other apps to use their own app for these tests.

Reply:
++

Reply:
Because this is an MVP to get us closer to 100% integration test coverage. In the original PR body, and even in the code, I had a `TODO: allow a configurable application construction`, which I removed -- it's not as trivial as you think btw. 

I'd rather not waste time on devising a near-perfect solution. An issue will be created to add such functionality at a later point in time.

Reply:
I'll bite the bullet and just do it.

Reply:
I’m fine with tackling this separately actually if that’s easier. From my quick read of `Config` it looks feasible. Not sure what complexities may arise though